_overviews/scala3-book/fp-pure-functions.md

```diff
@@ -42,23 +38,23 @@ These `String` methods are also pure functions:
 
 Most methods on the Scala collections classes also work as pure functions, including `drop`, `filter`, `map`, and many more.
 
-> In Scala, *functions* and *methods* are almost completely interchangeable, so even though we use the common industry term “pure function,” this term can be used to describe both functions and methods.
+> In Scala, _functions_ and _methods_ are almost completely interchangeable, so even though we use the common industry term “pure function,” this term can be used to describe both functions and methods.
 > If you’re interested in how methods can be used like functions, see the [Eta Expansion][eta] discussion.
 
 
 
 ## Examples of impure functions
 
-Conversely, the following functions are *impure* because they violate the definition.
+Conversely, the following functions are _impure_ because they violate the definition.
 
 The `foreach` method on collections classes is impure because it’s only used for its side effects, such as printing to STDOUT.
```

Comment:
In fact, this is not true. It will probably be called with impure functions but `List(1,2,3).foreach { () }` is perfectly pure.

Reply:
Is there any way to reword this, or is it best to just delete the `foreach` sentence and the three bullet points after it?

Reply:
If you want to keep this discussion, here’s a stab at rewording it. FWIW, the main reason I like this discussion is that it helps people keep an eye out for methods that return `Unit`. I know it was something I never considered many years ago. Here’s the rewording:

While the `foreach` method on collections classes is technically pure, it’s typically used for side effects, such as printing to STDOUT.

> A great hint that `foreach` might be used for side effects is that it returns the type `Unit`.
> Because it doesn’t return anything, logically the only reason you ever call it is to achieve some side effect.
> Similarly, _any_ method that returns `Unit` is a candidate to be an impure function.

Reply:
Sorry for not following up earlier. I think your argument is perfectly sensible. Maybe it is easiest to actually say exactly that: "Keep an eye out for methods that return `Unit`. Since they do not return anything useful, this return type indicates that they are only called for their side effects"?